src/VisualStudio/Core/Def/Implementation/Remote/VisualStudioRemoteHostClientProvider.cs

```diff
@@ -51,8 +51,21 @@ private VisualStudioRemoteHostClientProvider(HostWorkspaceServices services)
             _lazyClient = new AsyncLazy<RemoteHostClient>(CreateHostClientAsync, cacheResult: true);
         }
 
-        private Task<RemoteHostClient> CreateHostClientAsync(CancellationToken cancellationToken)
-            => ServiceHubRemoteHostClient.CreateAsync(_services, cancellationToken);
+        private async Task<RemoteHostClient> CreateHostClientAsync(CancellationToken cancellationToken)
+        {
+            var client = await ServiceHubRemoteHostClient.CreateAsync(_services, cancellationToken).ConfigureAwait(false);
+
+            if (System.Diagnostics.Debugger.IsAttached)
```

Comment:
💡 We should probably have an environment variable opt-out for this, just in case someone needs it.

Reply:
like in an integration test?